src/main/java/com/hazelcast/remotecontroller/CloudManager.java

```diff
@@ -0,0 +1,436 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
+import okhttp3.MediaType;
+import okhttp3.ResponseBody;
+import okhttp3.Response;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.HttpUrl;
+import okhttp3.Call;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import net.lingala.zip4j.ZipFile;
+import org.python.jline.internal.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class CloudManager {
+    private static final OkHttpClient CLIENT = new OkHttpClient();
+    private static final Logger LOG = LogManager.getLogger(Main.class);
+    private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+    private static final int TIMEOUT_FOR_CLUSTER_STATE_WAIT_IN_MINUTES = 5;
+    private static final int RETRY_TIME_IN_SECOND = 10;
+    private static final int CLUSTER_TYPE_ID = 5; // Serverless cluster
+    private static final int CLOUD_PROVIDER_ID = 1; // aws
+    private static final int CLOUD_PROVIDER_REGION_ID = 4; // us-west-2
+    private static final String CLUSTER_PLAN = "SERVERLESS";
+    private final ObjectMapper mapper = new ObjectMapper();
+    private String baseUrl;
+    private String bearerToken;
+    private Call call;
+
+    public CloudManager() {
+    }
+
+    private static boolean isNullOrEmpty(String s) {
+        return s == null || s.isEmpty();
+    }
+
+    public void loginToCloudUsingEnvironment() throws CloudException {
+        String baseUrl = System.getenv("BASE_URL");
+        String apiKey = System.getenv("API_KEY");
+        String apiSecret = System.getenv("API_SECRET");
+        String nullEnvVariables = "";
+        if (isNullOrEmpty(baseUrl)) {
+            nullEnvVariables = nullEnvVariables.concat(" BASE_URL ");
+        }
+        if (isNullOrEmpty(apiKey)) {
+            nullEnvVariables = nullEnvVariables.concat(" API_KEY ");
+        }
+        if (isNullOrEmpty(apiSecret)) {
+            nullEnvVariables = nullEnvVariables.concat(" API_SECRET ");
+        }
+        if (nullEnvVariables.isEmpty()) {
+            loginToCloud(baseUrl, apiKey, apiSecret);
+        } else {
+            throw new CloudException("Not all required environment variables are set. These are not set ones: " + nullEnvVariables);
+        }
+    }
+
+    public void loginToCloud(String url, String apiKey, String apiSecret) throws CloudException {
+        baseUrl = url;
+        bearerToken = getBearerToken(apiKey, apiSecret);
+        if (bearerToken == null) {
+            throw new CloudException("Login failed");
+        }
+    }
+
+    public CloudCluster createCloudCluster(String hazelcastVersion, boolean isTlsEnabled) throws CloudException {
```

Comment:
Is it possible to establish a connection without tls?

Reply:
I am sure we cannot on the prod environment but in dev environment tls and nontls tests passed on my local, so yes.